server/src/main/java/com/google/musicanalysis/site/YoutubeServlet.java

```diff
@@ -73,9 +78,8 @@ protected Boolean isMusic(String topic) {
      * @param genreCount hash map of frequency count of each music genre
      * @param numVideos maximum number of videos to retrieve
      */
-    protected void updateMusicCount(String youtubeResBody, HashMap<String, Integer> genreCount, int numVideos) {
-        JsonObject jObject = JsonParser.parseString(youtubeResBody).getAsJsonObject();
-        JsonArray videos = jObject.getAsJsonArray("items");
+    protected void updateMusicCount(JsonObject youtubeJsonObj, HashMap<String, Integer> genreCount) {
+        JsonArray videos = youtubeJsonObj.getAsJsonArray("items");
```

Comment:
Why are you doing this in the function? Can't you request the first parameter be `JsonArray videos`? This will improve readability. Generally, data formatting should be done outside your helper functions.